### DISCUSSION TITLE
Pre-install plugins, like nf-amazon

### DISCUSSION THREAD

**Discussion by maxbates** [nextflow-io/nextflow#2465] [Q&A]
Hi, is there a way to pre-install specific plugins?

When we run a workflow, we do not want the nextflow container to have internet access.

I also don't want nextflow to update itself (i.e. I don't think `nextflow self-update` is appropriate), just install a set of plugins.

We build nextflow in its own docker container, independent of any workflow / config, but we could create a temporary config to aid the installation.

**Comment by pditommaso** [comment]
You can use the command below run a dummy script and force the plugins pull 

```
echo "true" | nextflow run - -plugins nf-amazon
```
